sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/MarkAssemblyWithAttributeBase.cs

```diff
@@ -46,7 +46,8 @@ protected sealed override void Initialize(SonarAnalysisContext context)
                     var requiredAttributeFound = cc.Compilation.Assembly.GetAttributes().Any(a => a.AttributeClass.Is(AttributeToFind));
                     if (!requiredAttributeFound)
                     {
-                        cc.ReportDiagnosticWhenActive(Diagnostic.Create(Rule, null));
+                        cc.ReportDiagnosticWhenActive(Diagnostic.Create(Rule, null,
+                            properties: cc.Compilation.ToAssemblyNameProperties()));
```

Comment:
This is currently impossible to make some extra tests on `Diagnostic` with the current `Verifier` class. As it would take too long to refactor it to work like `Mock` (i.e. split the current `VerifyAnalyzer` method into a `Setup` and a `Verify` and then adding some `Callback` one so we can provide more custom tests).